ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/diffbased/verkle/storage/VerkleWorldStateKeyValueStorage.java

```diff
@@ -42,34 +46,50 @@
 
 import org.apache.tuweni.bytes.Bytes;
 import org.apache.tuweni.bytes.Bytes32;
+import org.apache.tuweni.units.bigints.UInt256;
 
 public class VerkleWorldStateKeyValueStorage extends DiffBasedWorldStateKeyValueStorage
     implements WorldStateKeyValueStorage {
 
-  protected BonsaiFullFlatDbStrategy flatDbStrategy;
+  protected final FlatDbStrategyProvider flatDbStrategyProvider;
+  protected final StemPreloader stemPreloader;
+  protected MetricsSystem metricsSystem;
+  protected final DataStorageConfiguration dataStorageConfiguration;
 
   public VerkleWorldStateKeyValueStorage(
-      final StorageProvider provider, final MetricsSystem metricsSystem) {
+      final StorageProvider provider,
+      final StemPreloader stemPreloader,
+      final DataStorageConfiguration dataStorageConfiguration,
+      final MetricsSystem metricsSystem) {
     super(
-        provider.getStorageBySegmentIdentifiers(
-            List.of(
-                ACCOUNT_INFO_STATE, CODE_STORAGE, ACCOUNT_STORAGE_STORAGE, TRIE_BRANCH_STORAGE)),
+        provider.getStorageBySegmentIdentifiers(List.of(CODE_STORAGE, TRIE_BRANCH_STORAGE)),
         provider.getStorageBySegmentIdentifier(KeyValueSegmentIdentifier.TRIE_LOG_STORAGE));
-    this.flatDbStrategy =
-        new BonsaiFullFlatDbStrategy(metricsSystem, new CodeHashCodeStorageStrategy());
+    this.stemPreloader = stemPreloader;
+    this.metricsSystem = metricsSystem;
+    this.dataStorageConfiguration = dataStorageConfiguration;
+    this.flatDbStrategyProvider =
+        new VerkleFlatDbStrategyProvider(metricsSystem, dataStorageConfiguration);
+    flatDbStrategyProvider.loadFlatDbStrategy(composedWorldStateStorage);
   }
 
   public VerkleWorldStateKeyValueStorage(
-      final BonsaiFullFlatDbStrategy flatDbStrategy,
       final SegmentedKeyValueStorage composedWorldStateStorage,
-      final KeyValueStorage trieLogStorage) {
+      final KeyValueStorage trieLogStorage,
+      final StemPreloader stemPreloader,
+      final DataStorageConfiguration dataStorageConfiguration,
+      final MetricsSystem metricsSystem) {
     super(composedWorldStateStorage, trieLogStorage);
-    this.flatDbStrategy = flatDbStrategy;
+    this.metricsSystem = metricsSystem;
+    this.dataStorageConfiguration = dataStorageConfiguration;
+    this.flatDbStrategyProvider =
+        new VerkleFlatDbStrategyProvider(metricsSystem, dataStorageConfiguration);
+    flatDbStrategyProvider.loadFlatDbStrategy(composedWorldStateStorage);
```

Comment:
Why is that the FlatDbStrategyProvider not calling `loadFlatDbStrategy` ? Is the object of any use without anything loaded on it?

I actually found out then that the `getFlatDbStategy` needs the same argument as the load and I found that weird. After checking the getter I can see that the getter calls `loadFlatDbStrategy`. Any particular reason for that? Why can't we call load in the constructor and make sure `flatDbStrategy is never null inside? Do we need lazy initialization? Only if it makes sense to create a provider and not use it...

Reply:
updated it was needed before but I think we had some refactoring of this part and it's not needed anymore in bonsai and verkle, I updated the code